src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameHeaders.cs

```diff
@@ -232,6 +233,11 @@ public static void ValidateHeaderCharacters(string headerCharacters)
             }
         }
 
+        public static void ValidateLongValue(StringValues value)
+        {
+            long.Parse(value, NumberStyles.AllowLeadingWhite | NumberStyles.AllowTrailingWhite, CultureInfo.InvariantCulture);
```

Comment:
Might make things a little easier to debug if this was wrapped in an InvalidOperationException stating the Content-Length header must be a positive decimal number or something like this.